flow/project.py

```diff
@@ -1597,6 +1610,39 @@ def __init__(self, config=None, environment=None, entrypoint=None):
         self._group_to_aggregate_store = _bidict()
         self._register_groups()
 
+    def _check_flow_schema_compatibility(self):
+        """Check whether this project's data schema is compatible with this version.
+
+        Raises
+        ------
+        :class:`~flow.errors.IncompatibleSchemaVersion`
+            If the schema version is incompatible.
+
+        """
+        schema_version = version.parse(SCHEMA_VERSION)
+        config_schema_version = version.parse(
+            flow_config.get_config_value("schema_version", config=self.config)
+        )
+        if config_schema_version > schema_version:
+            # Project config schema version is newer and therefore not supported.
+            raise IncompatibleSchemaVersion(
+                "The signac-flow schema version used by this project is '{}', but flow {} "
+                "only supports up to schema version '{}'. Try updating flow.".format(
+                    config_schema_version, __version__, schema_version
+                )
+            )
+        elif config_schema_version < schema_version:
+            raise IncompatibleSchemaVersion(
+                "The flow schema version used by this project is '{}', but flow {} "
+                "requires schema version '{}'. Please use '$ flow migrate' to "
+                "irreversibly migrate this project's schema to the supported "
+                "version.".format(config_schema_version, __version__, schema_version)
```

Comment:
```suggestion
                "The signac-flow schema version used by this project is '{}', but signac-flow {} "
                "requires schema version '{}'. Please use '$ flow migrate' to "
                "irreversibly migrate this project's schema to a supported "
                "version.".format(config_schema_version, __version__, schema_version)
```

Reply:
Is it possible that the package will support multiple schema versions concurrently? That is, should this be able to say "requires 3 <= schema version <= 5" or similar?